common/utils.go

```diff
@@ -18,3 +23,83 @@ func ErrorMessages(errors []error) string {
 	}
 	return strings.Join(result, "\n")
 }
+
+func WithNetNS(ns netns.NsHandle, work func() error) error {
+	errChan := make(chan error)
+
+	go func() {
```

Comment:
Faulty reasoning. I was thinking that I didn't want to lock the thread servicing the calling goroutine, but that's bogus for two reasons - that particular goroutine needs to block until the work is done anyway, and other goroutines for other requests will automatically get shifted to another thread by the scheduler as long as max procs is high enough. I'll remove it.